src/main/java/uk/gov/ons/ssdc/supporttool/endpoint/CaseEndpoint.java

```diff
@@ -31,6 +38,60 @@ public CaseEndpoint(UserIdentity userIdentity, CaseService caseService) {
     this.caseService = caseService;
   }
 
+  @PostMapping("{caseId}/action/updateSensitiveField")
+  public ResponseEntity<?> updateSensitiveField(
+      @PathVariable(value = "caseId") UUID caseId,
+      @RequestBody UpdateSampleSensitive updateSampleSensitive,
+      @Value("#{request.getAttribute('userEmail')}") String userEmail) {
+
+    Case caze = caseService.getCaseByCaseId(caseId);
+
+    userIdentity.checkUserPermission(
+        userEmail,
+        caze.getCollectionExercise().getSurvey(),
+        UserGroupAuthorisedActivityType.UPDATE_SENSITIVE_SAMPLE);
+
+    Optional<List<String>> validationErrorsOpt =
+        validateFieldToUpdate(caze, updateSampleSensitive.getSampleSensitive());
+
+    if (validationErrorsOpt.isPresent()) {
+      String validatationErrorStr = String.join("|", validationErrorsOpt.get());
+      Map<String, String> body = Map.of("errors", validatationErrorStr);
+
+      return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }
+
+    caseService.buildAndSendUpdateSensitiveSampleEvent(updateSampleSensitive);
+
+    return new ResponseEntity<>(HttpStatus.OK);
+  }
+
+  private Optional<List<String>> validateFieldToUpdate(
```

Comment:
I don't think the optional is adding any benefit here. Empty list = zero errors = OK request, IMHO